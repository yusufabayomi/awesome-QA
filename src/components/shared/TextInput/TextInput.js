import PropTypes from 'prop-types';

const propTypes = {
  label: PropTypes.string.isRequired,
  input: PropTypes.object,
};

const TextInput = ({ label, input }) => {
  return (
    <>
      <label className='form-label'>{label}</label>
      <input data-testid='text' {...input} type='text' />
    </>
  );
};

TextInput.propTypes = propTypes;
export default TextInput;
