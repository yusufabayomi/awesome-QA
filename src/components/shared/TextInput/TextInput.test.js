import React from 'react';
import { render, cleanup } from '@testing-library/react';
import TextInput from './TextInput';

afterEach(cleanup);

describe('Renders TextInput Component', () => {
  const props = {
    label: 'Question',
  };

  it('renders correctly', () => {
    const { asFragment } = render(<TextInput {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders an input field with text atribute', () => {
    const { getByTestId } = render(<TextInput {...props} />);
    expect(getByTestId('text')).toHaveAttribute('type', 'text');
  });
});
