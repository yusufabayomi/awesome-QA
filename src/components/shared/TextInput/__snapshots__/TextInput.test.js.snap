// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders TextInput Component renders correctly 1`] = `
<DocumentFragment>
  <label
    class="form-label"
  >
    Question
  </label>
  <input
    data-testid="text"
    type="text"
  />
</DocumentFragment>
`;
